tsdb/index/inmem/inmem.go

```diff
@@ -274,6 +274,48 @@ func (i *Index) MeasurementTagKeysByExpr(name []byte, expr influxql.Expr) (map[s
 	return mm.TagKeysByExpr(expr)
 }
 
+// TagKeyHasAuthorizedSeries determines if there exists an authorized series for
+// the provided measurement name and tag key.
+func (i *Index) TagKeyHasAuthorizedSeries(auth query.Authorizer, name []byte, key string) bool {
+	i.mu.RLock()
+	mm := i.measurements[string(name)]
+	i.mu.RUnlock()
+
+	if mm == nil {
+		return false
+	}
+
+	// TODO(edd): This looks like it's inefficient. Since a series can have multiple
+	// tag key/value pairs on it, it's possible that the same unauthorised series
+	// will be checked multiple times. It would be more efficient if it were
+	// possible to get the set of unique series IDs for a given measurement name
+	// and tag key.
+	var authorized bool
+	mm.SeriesByTagKeyValue(key).Range(func(_ string, seriesIDs SeriesIDs) bool {
+		if auth == nil || auth == query.OpenAuthorizer {
```

Comment:
This check can be moved up to the top of this method. Also, since `OpenAuthorizer` is now just a public package level variable instead of a type, it should not be checked for equality. However unlikely, it could be reassigned a value that no longer just returns `true` for all its methods. This is my fault, anyway... I should have just allowed for a `nil` Authorizer everywhere initially to mean "bypass authorization checks", rather than trying to mock a valid authorizer.

Reply:
@joelegasse https://github.com/influxdata/influxdb/issues/9159 to track this.